test/unit-test/MQTTv5/core_mqttv5_serializer_utest.c

```diff
@@ -0,0 +1,3299 @@
+    MQTTConnectProperties_t properties;
+    MQTTStatus_t status;
+    MQTTUserProperties_t userProperty;
+
+    memset( &properties, 0x0, sizeof( properties ) );
+    memset( &packetInfo, 0x0, sizeof( packetInfo ) );
+    uint8_t buffer[ 200 ] = { 0 };
+    uint8_t * pIndexLocal = buffer;
+    buffer[ 0 ] = 0x01;
+    buffer[ 1 ] = 0x00;
+    bool session = false;
+    packetInfo.pRemainingData = buffer;
+    packetInfo.type = MQTT_PACKET_TYPE_CONNACK;
+    pIndexLocal = &buffer[ 2 ];
+    properties.pIncomingUserProperty = &userProperty;
+    size_t propertyLength = encodeRemainingLength( pIndexLocal, 3 );
+    packetInfo.remainingLength = propertyLength + 5;
+    pIndexLocal++;
+    *pIndexLocal = MQTT_RECEIVE_MAX_ID;
+    pIndexLocal++;
+    pIndexLocal[ 0 ] = UINT16_HIGH_BYTE( 13 );
+    pIndexLocal[ 1 ] = UINT16_LOW_BYTE( 13 );
+    pIndexLocal = &pIndexLocal[ 2 ];
+    status = MQTTV5_DeserializeConnack( &properties, &packetInfo, &session );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    TEST_ASSERT_EQUAL_INT( 13, properties.serverReceiveMax );
+
+    /*Receive Max cannot have a value 0*/
+    pIndexLocal = &buffer[ 2 ];
+    propertyLength = encodeRemainingLength( pIndexLocal, 3 );
+    packetInfo.remainingLength = propertyLength + 5;
+    pIndexLocal++;
+    *pIndexLocal = MQTT_RECEIVE_MAX_ID;
+    pIndexLocal++;
+    pIndexLocal[ 0 ] = UINT16_HIGH_BYTE( 0 );
+    pIndexLocal[ 1 ] = UINT16_LOW_BYTE( 0 );
+    pIndexLocal = &pIndexLocal[ 2 ];
+    status = MQTTV5_DeserializeConnack( &properties, &packetInfo, &session );
+    TEST_ASSERT_EQUAL_INT( MQTTProtocolError, status );
+
+    /*Protocol error to include the same property twice*/
+    pIndexLocal = &buffer[ 2 ];
+    propertyLength = encodeRemainingLength( pIndexLocal, 6 );
+    packetInfo.remainingLength = propertyLength + 8;
+    pIndexLocal++;
+    *pIndexLocal = MQTT_RECEIVE_MAX_ID;
+    pIndexLocal++;
+    pIndexLocal[ 0 ] = UINT16_HIGH_BYTE( 13 );
+    pIndexLocal[ 1 ] = UINT16_LOW_BYTE( 13 );
+    pIndexLocal = &pIndexLocal[ 2 ];
+    *pIndexLocal = MQTT_RECEIVE_MAX_ID;
+    pIndexLocal++;
+    pIndexLocal[ 0 ] = UINT16_HIGH_BYTE( 13 );
+    pIndexLocal[ 1 ] = UINT16_LOW_BYTE( 13 );
+    pIndexLocal = &pIndexLocal[ 2 ];
+    status = MQTTV5_DeserializeConnack( &properties, &packetInfo, &session );
+    TEST_ASSERT_EQUAL_INT( MQTTProtocolError, status );
+
+    /*Invalid property length*/
+    packetInfo.remainingLength = 5;
+    pIndexLocal = &buffer[ 2 ];
+    propertyLength = encodeRemainingLength( pIndexLocal, 2 );
+    pIndexLocal++;
+    *pIndexLocal = MQTT_RECEIVE_MAX_ID;
+    pIndexLocal++;
+    pIndexLocal[ 0 ] = UINT16_HIGH_BYTE( 13 );
+    pIndexLocal[ 1 ] = UINT16_LOW_BYTE( 13 );
+    pIndexLocal = &pIndexLocal[ 2 ];
+    status = MQTTV5_DeserializeConnack( &properties, &packetInfo, &session );
+    TEST_ASSERT_EQUAL_INT( MQTTMalformedPacket, status );
+}
+
+void test_MQTTV5_DeserializeConnackOnlyMaxQos( void )
+{
+    MQTTPacketInfo_t packetInfo;
```

Comment:
I think there is duplication of the first few lines of code across the deserialize tests. Can you verify and see if duplication exists and can be wrapped in an helper function?

Reply:
The code code and initialisation for each test case can be moved to "void setUp( void )"
Which is use to setup for each test case.

Reply:
Check[ this for example](https://github.com/awslabs/amazon-kinesis-video-streams-rtp/blob/main/test/unit-test/packetization/packetization_utest.c#L23)